src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEMKernel.hpp

```diff
@@ -0,0 +1,723 @@
+    for( localIndex i = 0; i < 3; ++i )
+    {
+      // init at (-1,-1,-1) as the origin of the referential elem
+      coordsOnRefElem[i] = -1.0;
+      for( localIndex j = 0; j < 3; ++j )
+      {
+        coordsOnRefElem[i] += invJ[i][j] * (coords[j] - xLocal[0][j]);
+      }
+    }
+  }
+
+  GEOSX_HOST_DEVICE
+  static real64
+  evaluateRicker( real64 const & time_n,
+                  real64 const & f0,
+                  localIndex const & order )
+  {
+    real64 const o_tpeak = 1.0/f0;
+    real64 pulse = 0.0;
+    if((time_n <= -0.9*o_tpeak) || (time_n >= 2.9*o_tpeak))
+    {
+      return pulse;
+    }
+
+    constexpr real64 pi = M_PI;
+    real64 const lam = (f0*pi)*(f0*pi);
+
+    switch( order )
+    {
+      case 2:
+      {
+        pulse = 2.0*lam*(2.0*lam*(time_n-o_tpeak)*(time_n-o_tpeak)-1.0)*exp( -lam*(time_n-o_tpeak)*(time_n-o_tpeak));
+      }
+      break;
+      case 1:
+      {
+        pulse = -2.0*lam*(time_n-o_tpeak)*exp( -lam*(time_n-o_tpeak)*(time_n-o_tpeak));
+      }
+      break;
+      case 0:
+      {
+        pulse = -(time_n-o_tpeak)*exp( -2*lam*(time_n-o_tpeak)*(time_n-o_tpeak) );
+      }
+      break;
+      default:
+        GEOSX_ERROR( "This option is not supported yet, rickerOrder must be 0, 1 or 2" );
+    }
+
+    return pulse;
+  }
+  /**
+   * @brief Launches the precomputation of the source and receiver terms
+   * @tparam EXEC_POLICY execution policy
+   * @tparam FE_TYPE finite element type
+   * @param[in] size the number of cells in the subRegion
+   * @param[in] numFacesPerElem number of face on an element
+   * @param[in] X coordinates of the nodes
+   * @param[in] elemGhostRank array containing the ghost rank
+   * @param[in] elemsToNodes map from element to nodes
+   * @param[in] elemsToFaces map from element to faces
+   * @param[in] elemCenter coordinates of the element centers
+   * @param[in] faceNormal array containing the normal of all faces
+   * @param[in] faceCenter array containing the center of all faces
+   * @param[in] sourceCoordinates coordinates of the source terms
+   * @param[out] sourceIsLocal flag indicating whether the source is local or not
+   * @param[out] sourceNodeIds indices of the nodes of the element where the source is located
+   * @param[out] sourceConstantsx constant part of the source terms in x-direction
+   * @param[out] sourceConstantsy constant part of the source terms in y-direction
+   * @param[out] sourceConstantsz constant part of the source terms in z-direction
+   * @param[in] receiverCoordinates coordinates of the receiver terms
+   * @param[out] receiverIsLocal flag indicating whether the receiver is local or not
+   * @param[out] receiverNodeIds indices of the nodes of the element where the receiver is located
+   * @param[out] receiverNodeConstants constant part of the receiver term
+   * @param[out] sourceValue array containing the value of the time dependent source (Ricker for e.g)
+   * @param[in] dt time-step
```

Comment:
Consider grouping the input parameters.

Reply:
Would it be meaningful to create an Class or Structure to handle the source parameters and computations?

Reply:
Same thing for the source: it will be refactor and this can indeed be a question (even if WaveSolverBase is supposed to handle the generic stuff for all wave solvers in particular the sources)